lib/tail-file.js

```diff
@@ -68,6 +69,24 @@ class TailFile extends Readable {
       }
       throw err
     }
+    if (startPos !== null && startPos !== undefined) {
+      if (typeof startPos !== 'number') {
+        const err = new TypeError('startPos must be an integer > 0')
+        err.code = 'ESTARTPOS'
+        err.meta = {
+          got: typeof startPos
+        }
+        throw err
+      }
+      if (startPos < 0 || !Number.isInteger(startPos)) {
+        const err = new TypeError('startPos must be an integer > 0')
+        err.code = 'ESTARTPOS'
+        err.meta = {
+          got: startPos
+        }
+        throw err
+      }
+    }
```

Comment:
Can these just be parts of the same if? they do the same thing.

```javascript
if(typeof startpos !== 'number' || (startpos < 0 ....) )
```

Reply:
they're slightly different, enough to warrant them being separate.  One is a type check, and the other is a value check, and the `meta` value is different from both.  It's simpler to read separately, even though they appear to be the same.

Reply:
Ah, there we go, I missed the `meta` difference.